portfolio/src/main/webapp/js/comments.js

```diff
@@ -0,0 +1,150 @@
+            <p class="rating">` + rating + `</p>
+            <div class="downvote-btn" type="button"></div>
+        </div>
+        <img class="delete-btn" src="images/delete-btn4.png">
+        <div class="comment-content">
+            <p class="comment-text">` + text + `</p>
+            <p class="comment-date">` + date + `</p>
+        </div>`;
+
+    // add event listeners for delete and vote buttons
+    liElement.querySelector('.delete-btn').addEventListener('click', function() {
+        deleteComment(commentID);
+    });
+    liElement.querySelector('.upvote-btn').addEventListener('click', () => {
+        voteComment(true, commentID)
+    });
+    liElement.querySelector('.downvote-btn').addEventListener('click', () => {
+        voteComment(false, commentID)
+    });
+    return liElement;
+}
+
+// add event listener to the logo
+document.querySelector('.nav-logo').addEventListener('click', function () {
+    document.location.href = '/';
+});
+
+
+// comments sort by selector
+for (const dropdown of document.querySelectorAll(".sort-type-select-wrapper")) {
+    dropdown.addEventListener('click', function () {
+        this.querySelector('.sort-type-select').classList.toggle('open');
+    })
+}
+
+for (const option of document.querySelectorAll(".sort-option")) {
+    if (option.attributes['data-value'].value === selectedSortType) {
+        option.parentNode.querySelector('.sort-option.selected').classList.remove('selected');
+        option.classList.add('selected');
+        option.closest('.sort-type-select').querySelector('.sort-type-select-trigger span').textContent = option.textContent;
+    }
+
+    option.addEventListener('click', function () {
+        if (!this.classList.contains('selected')) {
+            this.parentNode.querySelector('.sort-option.selected').classList.remove('selected');
+            this.classList.add('selected');
+            this.closest('.sort-type-select').querySelector('.sort-type-select-trigger span').textContent = this.textContent;
+        }
+        const type = this.attributes['data-value'].value;
+        selectedSortType = type;
+        localStorage.setItem('sortType', selectedSortType);
+        getComments(type);
+    })
+}
+
+// upvote and downvote buttons event listeners
+for (const btn of document.querySelectorAll(".upvote-btn")) {
+    btn.addEventListener('click', function () {
+        const commentID = this.parentNode.parentNode.id;
+        voteComment(true, commentID);
+    })
+}
+
+for (const btn of document.querySelectorAll(".downvote-btn")) {
+    btn.addEventListener('click', function () {
+        const commentID = this.parentNode.parentNode.id;
+        voteComment(false, commentID);
+    })
+}
+
+// event listener for the delete comment button
+for (const btn of document.querySelectorAll(".delete-btn")) {
+    btn.addEventListener('click', function() {
+        const commentID = this.parentNode.id;
+        deleteComment(commentID);
+    })
+}
```

Comment:
These look very similar. Maybe create a function that receives a selector and a callback function and call it 3 times?